src/org/zaproxy/zap/extension/ascanrulesAlpha/TestNoSQLInjection.java

```diff
@@ -0,0 +1,651 @@
+										String name = getName() + " - " + nosql.getName();
+										
+										bingo(Alert.RISK_HIGH, Alert.CONFIDENCE_HIGH, name, getDescription(), null, 
+												param, valueInjected, extraInfo,  getSolution(), injectedMsg);
+										break;
+									}
+								}
+								
+							} catch (SocketException ex) {
+		
+								log.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() + 
+										" when accessing: " + msg.getRequestHeader().getURI().toString());
+								continue;
+							}
+							finally {
+								if(isStop())
+									return;
+							}						
+						}
+
+					}
+					
+					/**
+					 * TODO is content-type json format?
+					 * 
+					 */
+					boolean jsonEnable = false;
+					if (jsonEnable) {
+						
+						for(String[] jpv : nosql.getJsonParamValueInjection()) {
+							
+							onlyExactMatch = false;
+							attackType = JSON_ATTACK;
+							injectedMsg = getNewMsg();
+							
+							
+							/**
+							 * TODO  set post request with json content-type
+							 * 
+							 */
+							injectedMsg.getRequestHeader().setHeader(HttpRequestHeader.CONTENT_TYPE, 
+									"application/json");
+							injectedMsg.getRequestHeader().setMethod(HttpRequestHeader.POST);
+							
+							try {
+								valueInjected = getParamJsonString(param, jpv);
+								setParameter(injectedMsg, param, valueInjected);
+								sendAndReceive(injectedMsg, false);
+								if(isBingo(baseMsg, injectedMsg, param, valueInjected, attackType, nosql, 
+										onlyExactMatch)) {
+									break;
+								}
+							}catch(SocketException e) {
+								log.debug("Caught " + e.getClass().getName() + " " + e.getMessage() + 
+								" when accessing: " + msg.getRequestHeader().getURI().toString());
+								continue;
+							}catch(JSONException e) {
+								log.debug("Caught " + e.getClass().getName() + " " + e.getMessage() + 
+								" when try to convert to json the value: " + jpv);
+								continue;
+							}
+							finally {
+								if(isStop())
+									return;
+							}
+						}
+					}
+	    		}
+			}
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+
+	private boolean isTimedInjected(long intervalBaseMessage, long intervalInjectedMessage, int sleep) {
+		long diff=intervalInjectedMessage-intervalBaseMessage;
+		if(diff>0 && (diff>=sleep-DELTA_TIME))
+			return true;
+		return false;
+	}
+
+
+	/**
+	 * 
+	 * @param originalMsg
+	 * @param injectedMsg
+	 * @param param
+	 * @param valueInj
+	 * @param attack
+	 * @param nosql
+	 * @return true if is discovered a vulnerability, false otherwise.
+	 */
+	private boolean isBingo(HttpMessage originalMsg, HttpMessage injectedMsg, String param, String valueInj, 
+			String attack, NOSQLDB nosql, boolean onlyExactMatch) {
+		
+		String originalText = originalMsg.getResponseBody().toString();
+		String injectedText = injectedMsg.getResponseBody().toString();
+		String name = getName() + " - " + nosql.getName();
+		String extraInfo = null;
+		
+

```

Comment:
Throughout this class there seems to be a ton of whitespace lines that just add to length without really improving readability. They could be removed.